lib/controller.rb

```diff
@@ -1,23 +1,28 @@
+require_relative 'game_factory.rb'
+
 class Controller
-  attr_reader :game
-  def initialize(game, display)
-    @game = game
+  attr_reader :new_game
+
+  def initialize(display, game_factory)
     @display = display
+    @game_factory = game_factory
   end
 
-  def main_game
-    @game.over?
+  def main_game(squares = [1, 2, 3, 4, 5, 6, 7, 8, 9])
+    game_setup(squares)
```

Comment:
Could you just pass in a game - as squares are immediately used to create a game.

Reply:
Good point. I'll change this